validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ExternalValidatorSource.java

```diff
@@ -101,11 +117,29 @@ public DeleteKeyResult deleteValidator(final BLSPublicKey publicKey) {
   }
 
   @Override
-  public AddLocalValidatorResult addValidator(
-      final KeyStoreData keyStoreData, final String password, final BLSPublicKey publicKey) {
+  public AddValidatorResult addValidator(
+      KeyStoreData keyStoreData, String password, BLSPublicKey publicKey) {
     throw new UnsupportedOperationException();
   }
 
+  @Override
+  public AddValidatorResult addValidator(final BLSPublicKey publicKey, URL signerUrl) {
+    if (!canUpdateValidators()) {
+      return new AddValidatorResult(
+          PostKeyResult.error("Cannot add validator to a read only source."), Optional.empty());
+    }
+
+    try {
+      final ValidatorProvider provider =
+          new ExternalValidatorSource.ExternalValidatorProvider(spec, publicKey);
+      externalValidatorSourceMap.put(publicKey, signerUrl);
+      return new AddValidatorResult(PostKeyResult.success(), Optional.of(provider.createSigner()));
```

Comment:
So one detail we'll need to ensure we get right is ensuring that the created validator winds up being wrapped with slashing protection.  We might deal with that below but noting it now so I don't forget because it's very important. :)